### ISSUE TITLE
geodist

### ISSUE THREAD

**Issue by Zhangyanghao520** [nicolasdonati/DUO-FM#6]
Hello, could you please provide the files in the deogist folder?

**Comment by nicolasdonati** [comment]
Hi, Thanks for the interest :)
I cannot upload geodist files because they are too heavy.
I used matlab code from here https://github.com/llorz/SGA19_zoomOut/blob/master/utils/%2BMESH/geodesics_to_all.m
but I will try to upload python code for this in coming weeks.

**Comment by zzifer** [comment]
Hi,May I ask when the file deogist is uploaded

**Comment by nicolasdonati** [comment]
Hi,
A python code to generate geodesic distance matrices is available here:
https://github.com/RobinMagnet/pyFM/blob/6d3e59a5155e040502115df6635dfa86f7b47c42/pyFM/mesh/geometry.py#L280
It is (partially) based on Mick Sharp's C++ implementation, which also has python bindings:
https://github.com/nmwsharp/potpourri3d

**Comment by zzifer** [comment]
think you for you reply,and i have a problem.Because scape training set is not provided in your code,i downloaded dataset scape from https://graphics.soe.ucsc.edu/private/data/SCAPE/.But an error occurred at runtime.
The error is: Process finished with exit code 139(interrupted by signal 11:SIGSEGV)
The reason for the error might be because I used the wrong training set.Could you check it for me。think you
<img width="622" alt="1669184935192" src="https://user-images.githubusercontent.com/52739675/203484609-67bb7e23-bcc3-41cc-8862-b96337f21392.png">
<img width="547" alt="image" src="https://user-images.githubusercontent.com/52739675/203484683-305d4715-113b-4249-919a-96984539ee98.png">
<img width="552" alt="1669184587175" src="https://user-images.githubusercontent.com/52739675/203483929-1eda8059-e671-4807-a9e8-371c6aaaf67f.png">

**Comment by zzifer** [comment]
The problem is solved after I convert the ply file to the off file.
But a new problem has arisen.
IndexError:index 20716 is out of bounds for axis 0 with size 1
- .-sad....55555555555555

**Comment by nicolasdonati** [comment]
I need more of the error to solve it.
The data I used is this: https://nuage.lix.polytechnique.fr/index.php/s/LJFXrsTG22wYCXx

**Comment by zzifer** [comment]
i love you~~~mua mua mua~~~ I can't run code on my current computer. I'll contact you tomorrow

**Comment by zzifer** [comment]
hi,Thanks for the dataset you provided. Now the program can run correctly.I guess the previous problem was that there was no correspondence between the corres file and the shapes_train file.
Here are some of my questions：
1.What do the a and r in the datasets scape_r and scape_a mean?
2.There seems to be a difference between the scape data set I got from https://ai.stanford.edu/~drago/Projects/scape/scape.html and the dataset you use. May I ask how these data have been processed?
3.I got the faust dataset from http://faust.is.tue.mpg.de/, but there are only ply, png and txt files in it. Do I need to convert txt to vts? Or what settings need to be modified?

**Comment by zzifer** [comment]
<img width="777" alt="cb7338196808fcb173eaf0c559c95bc" src="https://user-images.githubusercontent.com/52739675/203738055-63c39e05-9348-4275-b07e-9bd9363fa4d1.png">
<img width="501" alt="image" src="https://user-images.githubusercontent.com/52739675/203738759-5d1c059f-2fc0-4325-9d3a-5c451a9cac78.png">
<img width="508" alt="image" src="https://user-images.githubusercontent.com/52739675/203738907-c37523f8-7815-46b0-b897-33b71cc9836d.png">
<img width="416" alt="image" src="https://user-images.githubusercontent.com/52739675/203738979-32942cc8-ef62-46eb-b4c9-c894b0520116.png">
<img width="282" alt="image" src="https://user-images.githubusercontent.com/52739675/203739721-85358812-bcab-4a92-b00a-0d0f54387e10.png">

**Comment by nicolasdonati** [comment]
Ok so for FAUST and SCAPE original shapes, the ground-truth correspondence is 1 to 1 (basically the identity map).
For the shapes I provided: FAUST_r and SCAPE_r (r is for re-meshed, as in [here](https://github.com/llorz/SGA18_orientation_BCICP_code)), the re-meshing shuffles the vertices so we keep track of this in vts files, which we use to recover the ground-truth map.
The take-home message is that for original shapes, you can replace say X.vts by range(0, X.verts.shape[0]).

The _a datasets have been re-meshed by another method (it is detailed in the [paper](https://arxiv.org/abs/2204.13453.pdf), I think supplementary). _a stands for anisotropic as the purpose of these re-meshed shapes is to create difficult non-regular triangulation to show that some method overfit to triangulation information.

For the third question, I think it is answered in my first point :) I hope that helps !

**Comment by zzifer** [comment]
Thank you for your enthusiastic answer these days.I will contact you if I have any questions later.I wish you more publications at top conferences！！！
